lib/OpenQA/Parser/Format/XUnit.pm

```diff
@@ -76,8 +79,9 @@ sub parse {
                 my $content = '# Test messages ';
                 $content .= "# $tc->{name}\n" if $tc->{name};
 
-                for my $out ($tc->children('skipped, passed, error, failure')->each) {
-                    $tc_result = 'fail' if ($out->tag =~ m/failure|error/);
+                my %TC_RESULT_BY_TAG = (softfailure => 'softfail', failure => 'fail', error => 'fail');
```

Comment:
Please move this outside of the function. There's no need to initialize this hash each and every time this function is invoked. It would make sense to move it even outside the whole function.